django/db/backends/mysql/base.py

```diff
@@ -136,7 +137,10 @@ class DatabaseWrapper(BaseDatabaseWrapper):
         'tinyblob', 'blob', 'mediumblob', 'longblob', 'tinytext', 'text',
         'mediumtext', 'longtext', 'json',
     )
-
+    data_type_check_constraints = {
+        'PositiveIntegerField': '`%(column)s` >= 0',
+        'PositiveSmallIntegerField': '`%(column)s` >= 0',
+    }
```

Comment:
`data_type_check_constraints` should be added only when database `supports_column_check_constraints == True`, that's why one test is falling on MySQL. You can change this to a `cached_property`.